src/Microsoft.ComponentDetection.Detectors/pip/SimplePypiClient.cs

```diff
@@ -265,12 +269,14 @@ private async Task<HttpResponseMessage> RetryPypiRequestAsync(Uri uri, PipDepend
     /// <returns> Returns the httpresponsemessage. </returns>
     private async Task<HttpResponseMessage> GetPypiResponseAsync(Uri uri)
     {
+        await this.semaphore.WaitAsync();
```

Comment:
What kind of impact does this have on build times for large Python repositories?